tutor4points/core/views.py

```diff
@@ -289,9 +290,52 @@ def requests(request, id):
             form_request_response = RequestResponseForm()
     else:
         form_request_response = RequestResponseForm()
+    if request.method == "POST" and 'show-accepted' in request.POST:
+        requests_received = requests_received.filter(accepted=True)
+        requests_sent= requests_sent.filter(accepted=True)
+        accept_filter = "checked" 
+
+    if request.method == 'POST' and 'submit-rating' in request.POST: #code for rating, move once 'paid and done' functionality is added.
+        form_rating = RateTutorForm(request.POST)
+        if form_rating.is_valid():
+            rating = form_rating.save(commit=False)
+            userGivenTo = get_user_model().objects.get(pk=request.POST['request-tutor'])
+            rating.given_to = userGivenTo 
+            rating.given_by = request.user
+            rating.save()
+
+            # calcute new rating, and update the database
+            type = form_rating.cleaned_data['rating_type']
+
+            if type == 'tutor':#for tutors
+                tutor_ratings = Rating.objects.filter(given_to = userGivenTo).filter(rating_type = 'tutor')
+                count = 0
+                sum = 0
+                for i in tutor_ratings:
+                    count += 1
+                    sum += i.rating
+                userGivenTo.tutor_avg_rating = sum/count
+            else: #otherwise tutee
+                tutor_ratings = Rating.objects.filter(given_to = userGivenTo).filter(rating_type = 'tutee')
+                count = 0
+                sum = 0
+                for i in tutor_ratings:
+                    count += 1
+                    sum += i.rating
+                userGivenTo.tutee_avg_rating = sum/count
+            userGivenTo.save()
+
+            currentRequest = TutorRequest.objects.get(pk=request.POST['request-id'])
+            currentRequest.completed = True
```

Comment:
When a tutor marks a request as complete and submits a rating, the request will also disappear on the tutee's side before the tutee has submitted a rating (vice versa). I think we need to rethink how to check when a request is completed by both sides. One solution is to have a tutor_rating_completed and tutee_rating_completed for each request and then we can change completed to true when both tutor_rating_completed and tutee_rating_completed are true. Not sure if this is the best method though.